src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoEnumTrailingCommaCheck.java

```diff
@@ -0,0 +1,129 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ *  Checks that enum definition does not contain a trailing comma.
+ *  Rationale: JLS allows trailing commas in arrays and enumerations, but does not allow
+ *  them in other locations. To unify the coding style, the use of trailing commas should
+ *  be prohibited.
+ * </p>
+ * <pre>
+ * enum Foo1 {
+ *   FOO,
+ *   BAR;
+ * }
+ * </pre>
+ * <p>
+ *  The check demands that there should not be any comma after last constant in
+ *  enum definition.
+ * </p>
+ * <pre>
+ * enum Foo1 {
+ *   FOO,
+ *   BAR, //violation
+ * }
+ * </pre>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NoEnumTrailingComma&quot;/&gt;
+ * </pre>
+ * <p>
+ * Which results in the following violations:
+ * </p>
+ * <pre>
+ * enum Foo1 {
+ *   FOO,
+ *   BAR; //OK
+ * }
+ * enum Foo2 {
+ *   FOO,
+ *   BAR //OK
+ * }
+ * enum Foo3 {
+ *   FOO,
+ *   BAR, //violation
+ * }
+ * enum Foo4 {
+ *   FOO, // OK
+ *   BAR, // violation
```

Comment:
I question the rationale of the `OK` line. Check is only looking at trailing commas (very last one), and will ignore this one. Doesn't adding `OK` make it seem like the check could be violating these and in some situation it could put a warning on them? To me that confuses me into thinking is "trailing comma" just mean the trailing comma on the end of the line instead of the end of the grouping.
There are a few examples like this.

Reply:
Donee